lib/SPIRV/SPIRVLowerSaddIntrinsics.cpp

```diff
@@ -147,17 +147,94 @@ void SPIRVLowerSaddIntrinsicsBase::replaceSaddSat(Function &F) {
   replaceSaddOverflow(*SaddO);
 }
 
+namespace {
+
+// This function for a given @IntegerBitWidth generates the following IR:
```

Comment:
We have [u_add_sat](https://registry.khronos.org/SPIR-V/specs/1.0/OpenCL.ExtendedInstructionSet.100.html#u_add_sat) OpenCL extended instruction, why are we not using it and instead emitting its implementation?

Reply:
There is a unclear question whether we can rely on having this extended instruction set and what should we do if it isn't available.

Reply:
The translator project is 99% focused on `Compute` capabilities of SPIR-V, which are used by OpenCL. Any conformant OpenCL implementation must support `OpenCL.std` extended instruction set, see [The OpenCL™ SPIR-V Environment Specification](https://registry.khronos.org/OpenCL/specs/2.2/html/OpenCL_Env.html#_extended_instruction_sets)

Moreover, that extended instruction set is already extensively used by the translator and I doubt that you are targeting SPIR-V consumer which doesn't support it, because you would have already faced some issues.

Therefore, I would like us to continue to use it whenever possible and not to manually implement all LLVM intrinsics instead.

Reply:
Here is the new PR: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1797